test/jdk/java/nio/Buffer/CharBufferAsCharSequenceTest.java

```diff
@@ -0,0 +1,332 @@
+ * @key randomness
+ * @library /test/lib
+ * @build jdk.test.lib.RandomFactory
+ * @summary tests the CharBuffer implementations behaving as CharSequence in various states (position, limit, offset)
+ * @run junit CharBufferAsCharSequenceTest
+ */
+public class CharBufferAsCharSequenceTest {
+
+    private static final Random RAND = RandomFactory.getRandom();
+    private static final int SIZE = RAND.nextInt(128, 1153);
+
+    private static char[] randomChars() {
+        char[] chars = new char[SIZE];
+        for (int i=0; i<SIZE; ++i) {
+            chars[i] = (char) RAND.nextInt();
+        }
+        return chars;
+    }
+
+    private static CharBuffer randomizeRange(CharBuffer cb) {
+        int mid = cb.capacity() >>> 1;
+        int start = RAND.nextInt(mid - 3); // from 0 to mid
+        int end = RAND.nextInt(mid + 3, cb.capacity()); // from mid to capacity
+        cb.position(start);
+        cb.limit(end);
+        return cb;
+    }
+
+    private static void populateAndAddCases(String type, CharBuffer cb, List<Arguments> cases) {
+        assert cb.position() == 0 && cb.limit() == cb.capacity();
+        char[] buf = randomChars();
+        cb.put(buf);
+        cb.clear();
+        addCases(type, buf, cb, cases);
+    }
+
+    private static void addCases(String type, char[] buf, CharBuffer cb, List<Arguments> cases) {
+        assert cb.position() == 0 && cb.limit() == cb.capacity();
+        cases.add(Arguments.of(cb, buf, 0, buf.length, type + " full"));
+
+        CharBuffer rndRange = randomizeRange(cb.duplicate());
+        cases.add(Arguments.of(rndRange, buf, rndRange.position(), rndRange.limit(), type + "  at " + rndRange.position() + " through " + rndRange.limit()));
+        cases.add(Arguments.of(rndRange.slice(), buf, rndRange.position(), rndRange.limit(), type + " sliced at " + rndRange.position() + " through " + rndRange.limit()));
+
+        CharBuffer rndSlicedRange = randomizeRange(rndRange.slice());
+        cases.add(Arguments.of(rndSlicedRange, buf, rndRange.position() + rndSlicedRange.position(), rndRange.position() + rndSlicedRange.limit(), type + " sliced at " + rndRange.position() + " with position " + rndSlicedRange.position() + " and limit " + rndSlicedRange.limit()));
+    }
+
+    static List<Arguments> charBufferArguments() {
+        List<Arguments> args = new ArrayList<>();
+
+        populateAndAddCases("HeapCharBuffer", CharBuffer.allocate(SIZE), args);
+        populateAndAddCases("BEHeapByteBuffer", ByteBuffer.allocate(SIZE*2).order(ByteOrder.BIG_ENDIAN).asCharBuffer(), args);
+        populateAndAddCases("LEHeapByteBuffer", ByteBuffer.allocate(SIZE*2).order(ByteOrder.LITTLE_ENDIAN).asCharBuffer(), args);
+        populateAndAddCases("BEDirectByteBuffer", ByteBuffer.allocateDirect(SIZE*2).order(ByteOrder.BIG_ENDIAN).asCharBuffer(), args);
+        populateAndAddCases("LEDirectByteBuffer", ByteBuffer.allocateDirect(SIZE*2).order(ByteOrder.LITTLE_ENDIAN).asCharBuffer(), args);
```

Comment:
Having BE or LE at the start of the type name is a bit confusing, would be clear (and more like charset names) to append rather than prepend.